crates/aptos/src/move_tool/mod.rs

```diff
@@ -6,50 +6,122 @@
+            MoveTool::Test(tool) => to_common_result(tool.execute().await),
         }
     }
 }
 
+/// Compiles a package and returns the [`ModuleId`]s
+#[derive(Parser)]
+pub struct CompilePackage {
+    #[clap(flatten)]
+    move_options: MovePackageDir,
+}
+
+impl CompilePackage {
+    pub async fn execute(&self) -> Result<Vec<String>, Error> {
+        let build_config = BuildConfig {
+            generate_abis: false,
+            generate_docs: true,
+            install_dir: None,
+            ..Default::default()
+        };
+        let compiled_modules = compile_move(
+            build_config,
+            self.move_options.package_dir.as_path(),
+        )?;
+        // TODO: This can be serialized once move is updated
+        let mut ids = Vec::new();
+        compiled_modules
+            .iter()
+            .for_each(|module| ids.push(module.self_id().to_string()));
+        Ok(ids)
+    }
+}
+
+/// Run Move unit tests against a package path
 #[derive(Parser)]
-pub struct MoveCli {
+pub struct TestPackage {
     #[clap(flatten)]
-    move_args: move_cli::Move,
-    #[clap(subcommand)]
-    command: move_cli::Command,
+    move_options: MovePackageDir,
 }
 
-impl MoveCli {
-    fn execute(self) -> CliResult {
-        let error_descriptions: ErrorMapping =
-            bcs::from_bytes(cached_framework_packages::error_map())
-                .map_err(|err| err.to_string())?;
-        move_cli::run_cli(
-            aptos_vm::natives::aptos_natives(),
-            &INITIAL_COST_SCHEDULE,
-            &error_descriptions,
-            &self.move_args,
-            &self.command,
+impl TestPackage {
+    pub async fn execute(&self) -> Result<&'static str, Error> {
+        let config = BuildConfig {
+            test_mode: true,
+            install_dir: Some(tempfile::tempdir().unwrap().path().to_path_buf()),
+            ..Default::default()
+        };
+        // TODO: Use move-package instead of move-cli
+        let result = run_move_unit_tests(
+            self.move_options.package_dir.as_path(),
+            config,
+            UnitTestingConfig::default_with_bound(Some(100_000)),
+            aptos_natives(),
+            false,
         )
-        .map(|_| "".to_string())
-        .map_err(|err| err.to_string())
+        .map_err(|err| Error::UnexpectedError(err.to_string()))?;
+
+        // TODO: commit back up to the move repo
+        match result {
+            UnitTestResult::Success => Ok("Success"),
+            UnitTestResult::Failure => Ok("Failure"),
+        }
     }
 }
+
+/// Compiles a Move package dir, and returns the compiled modules.
+fn compile_move(
+    build_config: BuildConfig,
+    package_dir: &Path,
+) -> Result<Vec<CompiledModule>, Error> {
+    // TODO: Add caching
+    let package = build_config
+        .compile_package(package_dir, &mut Vec::new())
+        .map_err(|err| Error::MoveCompiliationError(err.to_string()))?;
+
+    let modules = package.compiled_modules();
```

Comment:
Why isn't this module sufficient? Could we add a doc block here. I ask because we have the move-samples and there's a few differences that aren't intuitive to me. Not that much of this is intuitive :).

Reply:
Yeah, mostly that there's no control over the output which leaves us to whatever the base move uses, but if we're fine with it we can use the move cli versions for everything

Reply:
> but if we're fine with it we can use the move cli versions for everything

Which is sounds like we aren't completely fine with it based on feedback :)

Reply:
This is stuff it looks like I might have pulled out of looking too deep in the CLI, I've dropped it all for now for the next PR when I push the compiled modules.